api/maven-api-core/src/main/java/org/apache/maven/api/services/Source.java

```diff
@@ -20,17 +20,35 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.file.Path;
 
 import org.apache.maven.api.annotations.Experimental;
+import org.apache.maven.api.annotations.Nonnull;
+import org.apache.maven.api.annotations.Nullable;
 
 /**
- * The source for a project's XML model.
+ * The source for parsing an object.
  *
  * @since 4.0.0
  */
 @Experimental
 public interface Source {
+
+    /**
+     * Provides access the file to be parsed, if this source is backed by a file.
+     */
+    @Nullable
+    Path getPath();
+
+    /**
+     * Creates a readable stream.
+     */
+    @Nonnull
     InputStream getInputStream() throws IOException;
 
+    /**
+     * Retrieves the location for this source, useful to display to the user.
```

Comment:
How does this differ from the path? What is a location if not  a path?

Reply:
The location could be a URL.  The point is that `getPath()` returns a path, but may be `null` if the source is created with an `InputStream`.